lib/imap.php

```diff
@@ -84,9 +92,26 @@ public function checkPassword($uid, $password) {
 
 		if($canconnect) {
 			$uid = mb_strtolower($uid);
-			$this->storeUser($uid);
+			$this->storeUser($uid, $pieces[1]);
 			return $uid;
 		}
 		return false;
 	}
+
+	protected function storeUser($uid, $group) {
```

Comment:
the first part of this function is identical to what we have in the parent class in base.php...
Is there a way to inherit this part from the parent class?

Reply:
cc @ChristophWurst @nickvergessen ?

Reply:
it's obviously not the right way to do it because I get this in my debug log:
`Declaration of OC_User_IMAP::storeUser($uid, $group) should be compatible with OCA\user_external\Base::storeUser($uid) at .../apps/user_external/lib/imap.php#22`
any help appreciated :)

Reply:
the signature has to be 100% the same. either add that parameter to the base as well or restructure the code. maybe inheritance isn't right here. [have you considered using composition? :wink: ](https://en.wikipedia.org/wiki/Composition_over_inheritance)

Reply:
thanks for the indication! I'll look into it... unfortunately my OOP skills are still very limated... lots of things to learn here :)

Reply:
okay so I went for a completely different approach in e0b2626 which allows us to further extend this with things like #69...
So I now added it back to the parent class, take an array of groups as param and loop through this one to add the user to possibly multiple groups...